### PR TITLE
Bringup Xiaomi Mi 9T (davinci) to 20.0 from 17.1

### PR DESCRIPTION
**Pull request by GarciaLnk**
Builds but it needs testing, let me know before publishing if you decide to merge this.

### CODE REVIEW

File: Scripts/Common/Optimize.sh [hunk 1]
```diff
@@ -65,7 +65,8 @@ sed -i 's/zramsize=.*/zramsize=50%,max_comp_streams=8/' essential/mata/rootdir/e
 sed -i 's/zramsize=.*/zramsize=50%,max_comp_streams=4/' zte/axon7/rootdir/etc/fstab.qcom &>/dev/null || true;
 sed -i 's/zramsize=.*/zramsize=50%,max_comp_streams=8/' sony/tama-common/rootdir/etc/fstab.qcom &>/dev/null || true;
 #4GB+ (2GB+)
-sed -i 's/zramsize=.*/zramsize=50%,max_comp_streams=8/' google/redbull/fstab.hardware google/sunfish/fstab.hardware oneplus/msm8998-common/rootdir/etc/fstab.qcom oneplus/sdm845-common/rootdir/etc/fstab.qcom xiaomi/sm6150-common/rootdir/etc/fstab*.qcom &>/dev/null || true;
+sed -i 's/zramsize=.*/zramsize=50%,max_comp_streams=8/' google/redbull/fstab.hardware google/sunfish/fstab.hardware oneplus/msm8998-common/rootdir/etc/fstab.qcom oneplus/sdm845-common/rootdir/etc/fstab.qcom &>/dev/null || true;
+sed -i 's/zramsize=.*/zramsize=50%,max_comp_streams=8,zram_backingdev_size=512M/' xiaomi/sm6150-common/rootdir/etc/fstab.zram &>/dev/null || true;
```
Comment (SkewedZeppelin) [inline Scripts/Common/Optimize.sh:69]: why did you remove this?

File: Scripts/LineageOS-17.1/Patch.sh [hunk 2]
```diff
@@ -539,14 +539,6 @@ if enterAndClear "device/oppo/common"; then
 awk -i inplace '!/TARGET_RELEASETOOLS_EXTENSIONS/' BoardConfigCommon.mk; #disable releasetools to fix delta ota generation
 fi;
 
-if enterAndClear "device/xiaomi/davinci"; then
-smallerSystem;
```
Comment (SkewedZeppelin) [inline Scripts/LineageOS-17.1/Patch.sh:543]: are you sure it fits?

Comment (GarciaLnk) [reply Scripts/LineageOS-17.1/Patch.sh:543]: The system partition in davinci is 3.5 GiB large, so I don't see why the system image would need further reduction. I have used DivestOS in davinci in the past up to 19.1, without using smallerSystem, and I haven't faced any issues.

File: Scripts/Common/Optimize.sh (hunk 1 above)
Comment (GarciaLnk) [reply Scripts/Common/Optimize.sh:69]: I'm just respecting the device tree changes, zram_backingdev_size seems to enable zram writeback with a 512M limit.

Alternatively, the sed regex could be changed to keep additional changes and only modify the values of zramsize and max_comp_streams if that's the intent of the patch.

**Comment by SkewedZeppelin** [comment]
Thank you for this, I'll merge this after I get the August ASB out for 20.0 after Lineage merges it.

**Comment by SkewedZeppelin** [comment]
Merged, thanks!

A davinci 20.0 build should be up on the server in a few minutes.
